File: go/general/257. 二叉树的所有路径.go
package main

import "strconv"

var paths []string

func binaryTreePaths(root *TreeNode) []string {
	paths = make([]string, 0)
	dfs(root, "")
	return paths
}

func dfs(root *TreeNode, path string) {
	if root != nil {
		path += strconv.Itoa(root.Val)
		if root.Left == nil && root.Right == nil {
			paths = append(paths, path)
		} else {
			path += "->"
			dfs(root.Left, path)
			dfs(root.Right, path)
		}
	}

}

File: python/树/257. 二叉树的所有路径.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

from typing import List
class Solution:
    def binaryTreePaths(self, root: TreeNode) -> List[str]:
        def dfs(root,s):
            if not root:
                return
            if not root.left and not root.right:
                s += str(root.val)
                return res.append(s)
            if root.left:
                dfs(root.left,s+str(root.val)+'->')
            if root.right:
                dfs(root.right,s+str(root.val)+'->')
        res = []
        dfs(root,'')
        return res


